mne/evoked.py

```diff
@@ -278,40 +277,7 @@ def decimate(self, decim, offset=0):
         self.times = self.times[decim_slice].copy()
         self.first = int(self.times[0] * self.info['sfreq'])
         self.last = len(self.times) + self.first - 1
-        return self
-
-    def shift_time(self, tshift, relative=True):
-        """Shift time scale in evoked data.
-
-        Parameters
-        ----------
-        tshift : float
-            The amount of time shift to be applied if relative is True
-            else the first time point. When relative is True, positive value
-            of tshift moves the data forward while negative tshift moves it
-            backward.
-        relative : bool
-            If true, move the time backwards or forwards by specified amount.
-            Else, set the starting time point to the value of tshift.
-
-        Returns
-        -------
-        evoked : instance of Evoked
-            The modified Evoked instance.
-
-        Notes
-        -----
-        Maximum accuracy of time shift is 1 / evoked.info['sfreq']
-        """
-        times = self.times
-        sfreq = self.info['sfreq']
-
-        offset = self.first if relative else 0
-
-        self.first = int(tshift * sfreq) + offset
-        self.last = self.first + len(times) - 1
-        self.times = np.arange(self.first, self.last + 1,
-                               dtype=np.float) / sfreq
+        # self._update_first_last()
```

Comment:
I would like to replace the two preceding lines in `decimate` with `_update_first_last()` but it breaks `mne/tests/test_evoked.py:test_decim()`, by changing the sample numbers.  Currently in the test, an Evoked w/ 20 timepoints (first=-10, last=9, times from -1 to 0.9) is decimated by a factor of 6, yielding first=-1, last=2, times=[-1, -0.4, 0.2, 0.8]. This means that "sample zero" was the sample associated with time=-0.4, which is *farther away from actual time zero* than the subsequent sample (the one associated with time 0.2).  So unless there is a convention that "sample zero" is always associated with the negative time in cases where no sample is exactly at time zero, I would argue that the test as it currently stands is wrong: if decimation yields times=[-1, -0.4, 0.2, 0.8], then "sample zero" should be the sample at time 0.2.  @larsoner WDYT?

Reply:
I would say yes that sample zero should be the one closest to 0.